### ISSUE TITLE
EXPLAIN (ANALYZE, PREFETCH) counters are wrong

### ISSUE THREAD

**Issue by hlinnaka** [neondatabase/neon#9199]
Hmm, the code was different before commit 78938d1b591b33d23495a0edb8b123cc5cac6a27. Before that, there was no `prefetch_register_bufferv` call in `neon_read_at_lsnv`, so a "synchronous" non-prefetched read would reach that `pgBufferUsage.prefetch.misses += 1;` line. After 78938d1b591b33d23495a0edb8b123cc5cac6a27, we always do `prefetch_register_bufferv` first, so you practically never reach that line.

So I believe we have a bug in how those counters are updated for EXPLAIN (PREFETCH), introduced in commit 78938d1b591b33d23495a0edb8b123cc5cac6a27. It only affects those numbers in EXPLAIN output, though, the logic is otherwise correct. I'll open a separate issue to track that.

_Originally posted by @hlinnaka in https://github.com/neondatabase/neon/pull/9190#discussion_r1780896623_

**Comment by ololobus** [comment]
@hlinnaka @knizhnik I see PR https://github.com/neondatabase/neon/pull/9272 is linked and merged. Is it fixed? Or do we need to do anything else?

**Comment by knizhnik** [comment]
Yes, this PR fixes the reported problem.
But reported prefetch statistic is rewritten several times since that commit.
Right now in main "prefetch hint" means that prefetch response was received before page is requested.
So now prefetch hits+misses can be much smaller than number of read pages.
But it doesn't means that prefetch is not working.

**Comment by hlinnaka** [comment]
Yeah, let's close this. There might be other things we don't like about the counters, but this bug has been fixed.
